### ISSUE TITLE
property:relationshipAccordingTo

### ISSUE THREAD

**Issue by nielsklazenga** [tdwg/tcs2#47]
### relationshipAccordingTo (property)

|    |    |
|----|----|
| Label | Relationship according to |
| Definition | Reference for the taxon relationship assertion |
| Usage notes |  |
| Comments |  |
| Required |  |
| Repeatable |  |
| Constraints |  |

**Comment by nielsklazenga** [comment]
I have been considering whether we should make this property not required (it is required at the moment), as it creates a loop when the TaxonConceptMapping is an assertion in an Annotation or Nano-publication, _e.g._:

```turtle
# Euphrasia gibbsiae sec. Barker 1982 > Euphrasia kingii sec. Stones & Curtis 1978
_:ann8 a oa:Annotation ;
    oa:motivatedBy oa:commenting ;
    oa:hasBody [ a tcs:TaxonConceptMapping ;
            tcs:mappingAccordingTo _:ann8 ;
            tcs:mappingRelation tcs:includes ;
            tcs:subjectTaxonConcept <https://id.biodiversity.org.au/instance/apni/545068> ;
            tcs:objectTaxonConcept _:tc7 ;
            rdfs:comment """Euphrasia gibbsiae sec. Barker 1982 also includes Euphrasia 
                    gibbsiae sec. Curtis & Stones 1973""" ] ;
    oa:hasTarget <https://id.biodiversity.org.au/instance/apni/545068> ;
    dcterms:creator <https://orcid.org/0000-0003-2224-6821> ;
    dcterms:created "2023-10-21T23:41:00+11" .
```

At the moment, it just makes the examples look ugly, so that is okay, but I think not being able to create TaxonConceptMappings inside Annotations or Nano-publications will be a problem in real-world situations. Moreover, I think the 'required' attribute only means something in the human-readable documentation and there we could also put in the usage that a source of the TaxonConceptMapping is required, but that you do not necessarily need to use the `mappingAccordingTo` property for it.

The same should probably also be considered with respect to the `accordingTo` property (#4) on the TaxonConcept.

What are other people's thoughts on this?

**Comment by nielsklazenga** [comment]
Thanks @camwebb , I've come to the same conclusion. I like the idea of inversion. In JSON-LD you can just do that without having to mint a new property. It does not work here, because it breaks the `oa:Annotation`, but it brings home that we are purely talking about limitations in he syntax here.

**Comment by jliljeblad** [comment]
Seems ok with me too. I'm a bit worried about imposing requirements in general unless you are sure that they actually are absolutely necessart.

**Comment by nielsklazenga** [comment]
I think it would be good to borrow `dcterms:creator` and `dcterms:created`. Then we can still keep `tcs:mappingAccordingTo`, but it does not need to be required. We can just add in the notes that a TaxonConceptMapping should have either `tcs:mappingAccordingTo` or `dcterms:creator` and `dcterms:created`, but this would be informative, not normative. I think this will make it much easier to make and manage these mappings (you do not have to keep an extra table with annotations for example).

**Comment by nielsklazenga** [comment]
These borrowed properties have now been added, as agreed on in the TCS 2 Task Group meeting on 4 December 2024.

**Comment by nielsklazenga** [comment]
Now that we've done that, I think it would be good to replace this property with `dcterms:source`.

**Comment by camwebb** [comment]
<http://purl.org/dc/terms/source> Definition: "A related resource from which the described resource is _derived_" (my italics). Is this definition sufficient for what we mean by `mappingAccodingTo`? Is a mapping _derived_ from its publication? - I think not.  But this is a semantic argument... If we do away with this one, we should do the same for a `accordingTo`.

**Comment by nielsklazenga** [comment]
Yes, _derived_ is not how I would have said it, but if you use _mappingAccordingTo_ it means that the mapping comes from that publication, so it is correct and exactly what we mean.

I do not think that if we replace `mappingAccordingTo` with `source` means that we also need to replace `accordingTo` with `source`, also given your earlier statements to the extent that `mappingAccordingTo` and `accordingTo` are entirely different beasts. `accordingTo` and `namePublishedIn` have special meaning in TCS beyond attribution, while `mappingAccordingTo` is purely attribution. A name with a different `namePublishedIn` is a different name and a taxon concept with a different `accordingTo` is a different taxon concept; a mapping with a different `mappingAccordingTo` is the same mapping. That's why I think that we can replace `mappingAccordingTo` with general Dublin Core terms like `source` or `creator` plus `created` but `accordingTo` and `namePublishedIn` we cannot. Even though they are the `source` of the taxon concept and taxon name respectively, that is not the only thing they are.

If we could do this in TDWG standards, `accordingTo` and `namePublishedIn` could be sub-properties of `source`. In the SSSOM mappings that the TAG seems to be pushing the following mappings would apply:
- `accordingTo skos:broadMatch dcterms:source`
- `namePublishedIn skos:broadMatch dcterms:source`
- `mappingAccordingTo skos:exactMatch dcterms:source`

**Comment by nielsklazenga** [comment]
I actually do not object to replacing `accordingTo` and `namePublishedIn` with `dcterms:source` (in fact, I quite like it), but, as I said above, I think it can be justified to keep `accordingTo` and `namePublishedIn` while replacing `mappingAccordingTo` with `dcterms:source`.

**Comment by nielsklazenga** [comment]
Just noting that we also have the `typeAccordingTo` term, which is used for lectotypifications and neotypifications (and maybe epitypification and conservation). This is probably the strongest argument to at least keep `accordingTo` and `namePublishedIn`, as replacing `typeAccordingTo` with `dcterms:source` would make it ambiguous as it could be any publication that lists a nomenclatural type, while at the same time we cannot just keep this term and replace all the others with `dcterms:source`. 

[Not sure if `typeAccordingTo` is any less ambiguous, but that is a discussion for another property.]

**Comment by nielsklazenga** [comment]
Another thing to note—and this was the rationale for my proposal to change `mappingAccordingTo` to `dcterms:source`—is that `accordingTo` and `namePublishedIn` are IRI equivalents of `dwc:nameAccordingTo` and `dwc:namePublishedIn` respectively. `mappingAccordingTo` was the IRI equivalent of `dwc:relationshipAccordingTo`, but is not anymore since we added `dcterms:creator`. So, we basically split `mappingAccordingTo` into `dcterms:source` and `dcterms:creator` (which is why I do not like `mappingAcordingTo` anymore). The Darwin Core `ResourceRelationship`, which we would have used for this if it could be used with RDF, also has a `relationshipEstablishedDate` property, for which I think `dcterms:created` is more appropriate in our context.

**Comment by nielsklazenga** [comment]
@ben-norton , I would not mind your input on this last issue, _i.e._ whether or not use `dcterms:source` instead of `tcs:mappingAccordingTo`. I prefer `dcterms:source` for this, because `mappingAccordingTo` is all about provenance and has no special meaning in TCS, so I think it is better to use a term from a library standard like Dublin Core than defining a term for it in TCS. In fact, I think `dcterms:creator`, `dcterms:created` and `dcterms:source` can be useful on a NomenclaturalType object as well.

**Comment by nielsklazenga** [comment]
I think we might keep `tcs:mappingAccordingTo` and make `dcterms:creator`, `dcterms:created` and `dcterms:source` record-level properties that can be used on different classes (if we have to borrow them at all). `dcterms:source` will be very useful for Nomenclatural  Types as well, as the information often comes from publications other than the publication in which the typification was made Likewise, when coming from the specimen (Darwin Core) side, one would want to use `dcterms:creator` and `dcterms:created` for type status annotations.
